src/figuresClasses.ts

```diff
@@ -1,11 +1,85 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
 
-export class Triangle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape = 'triangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Fail data!!!');
+    }
+
+    const arr: number[] = [a, b, c];
+
+    arr.sort((x: number, y: number) => x - y);
+
+    if (arr[2] >= arr[0] + arr[1]) {
+      throw new Error('Fail data!!!');
```

Comment:
[CODE STYLE] - The error message here should also be more descriptive. Consider something like 'The sum of any two sides of a triangle must be greater than the third side.'